packages/jetpack-connection/lib/Manager.php

```diff
@@ -0,0 +1,262 @@
+	/**
+	 * Returns true if the current site is connected to WordPress.com.
+	 *
+	 * @return Boolean is the site connected?
+	 */
+	public function is_active() {
+
+	}
+
+	/**
+	 * Returns true if the user with the specified identifier is connected to
+	 * WordPress.com.
+	 *
+	 * @param Integer $user_id the user identifier.
+	 * @return Boolean is the user connected?
+	 */
+	public function is_user_connected( $user_id ) {
+
+	}
+
+	/**
+	 * Get the wpcom user data of the current|specified connected user.
+	 *
+	 * @param Integer $user_id the user identifier.
+	 * @return Object the user object.
+	 */
+	public function get_connected_user_data( $user_id ) {
+
+	}
+
+	/**
+	 * Is the user the connection owner.
+	 *
+	 * @param Integer $user_id the user identifier.
+	 * @return Boolean is the user the connection owner?
+	 */
+	public function is_connection_owner( $user_id ) {
+
+	}
+
+	/**
+	 * Unlinks the current user from the linked WordPress.com user
+	 *
+	 * @param Integer $user_id the user identifier.
+	 */
+	public static function disconnect_user( $user_id ) {
+
+	}
+
+	/**
+	 * Initializes a transport server, whatever it may be, saves into the object property.
+	 * Should be changed to be protected.
+	 */
+	public function initialize_server() {
+
+	}
+
+	/**
+	 * Checks if the current request is properly authenticated, bails if not.
+	 * Should be changed to be protected.
+	 */
+	public function require_authentication() {
+
+	}
+
+	/**
+	 * Verifies the correctness of the request signature.
+	 * Should be changed to be protected.
+	 */
+	public function verify_signature() {
+
+	}
+
+	/**
+	 * Attempts Jetpack registration which sets up the site for connection. Should
+	 * remain public because the call to action comes from the current site, not from
+	 * WordPress.com.
+	 *
+	 * @return Integer zero on success, or a bitmask on failure.
+	 */
+	public function register() {
+
+	}
+
+	/**
+	 * Returns the callable that would be used to generate secrets.
+	 *
+	 * @return Callable a function that returns a secure string to be used as a secret.
+	 */
+	protected function get_secret_callable() {
+		if ( ! isset( $this->secret_callable ) ) {
+			/**
+			 * Allows modification of the callable that is used to generate connection secrets.
+			 *
+			 * @param Callable a function or method that returns a secret string.
+			 */
+			$this->secret_callable = apply_filters( 'jetpack_connection_secret_generator', 'wp_generate_password' );
+		}
+
+		return $this->secret_callable;
+	}
+
+	/**
+	 * Returns the object that is to be used for all option manipulation.
+	 *
+	 * @return Object $manager an option manager object.
+	 */
+	protected function get_option_manager() {
+		if ( ! isset( $this->option_manager ) ) {
+			/**
+			 * Allows modification of the object that is used to manipulate stored data.
+			 *
+			 * @param Jetpack_Options an option manager object.
+			 */
+			$this->option_manager = apply_filters( 'jetpack_connection_option_manager', false );
```

Comment:
Yes, there needs to be a sane default or better error handling in case the manager is not set. I don't think there's going to be a case when this happens at runtime, though - it's a matter of an improperly set up class.